lib/pbench/server/cache_manager.py

```diff
@@ -93,16 +93,19 @@ def __init__(self, path: Path):
         """
         self.name = path.name
         self.location = path
+        self.resolve_path = path.resolve(strict=True)
+        self.resolve_type = None
+        self.size = None
         if path.is_dir():
             self.type = "directory"
         elif path.is_symlink():
             self.type = "symlink"
-            self.resolve_path = path.readlink()
+            if self.resolve_path.is_symlink():
+                self.file_info(self.resolve_path)
             self.resolve_type = "directory" if self.resolve_path.is_dir() else "file"
         elif path.is_file():
-            self.name = path.name
-            self.size = path.stat().st_size
             self.type = "file"
+            self.size = path.stat().st_size
```

Comment:
if path is a symlink, it will be added as a symlink but there are entries like `resolve_path` and `resolve_type` which will be the only attributed which will be having info about the resolved path.

check of symlink inside a symlink block was intended for the edge case that you suggested: If even after resolving a symlink another symlink appear. So, I am just trying to resolve that.

if the resolved path is a symlink, then it will be again checked if it is a symlink and again will be parsed through `file_info`

Reply:
> if path is a symlink, it will be added as a symlink 

No, it won't -- not as the code currently stands -- if the path refers to a directory:  [`is_dir()`](https://docs.python.org/3/library/pathlib.html#pathlib.Path.is_dir) will return `True` if the path is a valid symlink referring to a directory.

So, we need to decide which is more important:  reporting it as a symlink or treating it as a directory.  Treating it as a directory is potentially dangerous and requires a bunch of defensive code which this PR doesn't include, so I would recommend checking for symlink's _first_ and _not_ treating them as directories.

> check of symlink inside a symlink block was intended for the edge case that you suggested: If even after resolving a symlink another symlink appear. So, I am just trying to resolve that.

Your use of [`resolve()`](https://docs.python.org/3/library/pathlib.html#pathlib.Path.resolve) (instead of `readlink()`) will chase the symlink chain for you, so we don't need to do that here.  (If we're not going to permit the traversal to follow directory symlinks, then we don't need to chase the chain, and the result of `resolve()` is sufficient.)  (So, I don't think we need `file_info()`.)